sentry-android-ndk/src/main/jni/sentry.c

```diff
@@ -11,6 +11,190 @@ struct transport_options {
+
+                        if (iterator) {
+                            jobject iter = (*env)->CallObjectMethod(env, set, iterator);
+
+                            if (iter) {
+                                jclass iteratorClass = (*env)->FindClass(env, "java/util/Iterator");
+
+                                if (iteratorClass) {
+                                    jmethodID hasNext = (*env)->GetMethodID(env, iteratorClass, "hasNext", "()Z");
+
+                                    if (hasNext) {
+                                        jmethodID next =
+                                                (*env)->GetMethodID(env, iteratorClass, "next", "()Ljava/lang/Object;");
+
+                                        if (next) {
+                                            jclass entryClass = (*env)->FindClass(env, "java/util/Map$Entry");
+                                            if (entryClass) {
+                                                jmethodID getKey =
+                                                        (*env)->GetMethodID(env, entryClass, "getKey", "()Ljava/lang/Object;");
+
+                                                jmethodID getValue =(*env)->GetMethodID(env, entryClass, "getValue", "()Ljava/lang/Object;");
+
+                                                // Iterate over the entry Set
+                                                while ((*env)->CallBooleanMethod(env, iter, hasNext)) {
+                                                    jobject entry = (*env)->CallObjectMethod(env, iter, next);
+                                                    jstring key = (jstring) (*env)->CallObjectMethod(env, entry, getKey);
+                                                    // TODO: value can be object e this would make it harder
+                                                    jstring value = (jstring) (*env)->CallObjectMethod(env, entry, getValue);
+                                                    const char* keyStr = (*env)->GetStringUTFChars(env, key, NULL);
+                                                    if (!keyStr) {  // Out of memory
+                                                        return;
+                                                    }
+                                                    const char* valueStr = (*env)->GetStringUTFChars(env, value, NULL);
+                                                    if (!valueStr) {  // Out of memory
+                                                        (*env)->ReleaseStringUTFChars(env, key, keyStr);
+                                                        return;
+                                                    }
+
+                                                    sentry_value_set_by_key(
+                                                            rvData, keyStr, sentry_value_new_string(valueStr));
+
+                                                    (*env)->DeleteLocalRef(env, entry);
```

Comment:
similarly, do we need this for `set`/`iter` as well? basically everything thats a `jobject`?

Reply:
mm probably, it won't hurt anyway.